ludwig/automl/automl.py

```diff
@@ -144,6 +157,65 @@ def auto_train(
     return train_with_config(dataset, config, output_directory=output_directory, random_seed=random_seed, **kwargs)
 
 
+def create_auto_config_with_dataset_profile(
+    target: str = None,
+    dataset: Optional[Union[str, DataFrame]] = None,
+    dataset_profile: dataset_profile_pb2.DatasetProfile = None,
+    random_seed: int = default_random_seed,
+    include_hyperopt: bool = False,
+    time_limit_s: Union[int, float] = None,
+    backend: Union[Backend, str] = None,
+) -> dict:
+    """Returns the best single-shot Ludwig config given a Ludwig dataset or dataset profile.
+
+    If only the dataset is provided, then a new profile is computed.
+    Only one of the dataset or dataset_profile should be specified, not both.
+
+    This function is intended to eventually replace create_auto_config().
+    """
+    if dataset is None and dataset_profile is None:
```

Comment:
Would it also make sense to add a check for `target` to make sure that it isn't None? or is that handled within the dataset_profile generation functions?

Reply:
Done.